From 9ffc462f4b36107f8d2063ae5f11b64ef7d68ef8 Mon Sep 17 00:00:00 2001
From: Artur Weber <aweber.kernel@gmail.com>
Date: Thu, 8 Jan 2026 17:28:32 +0100
Subject: [PATCH] Use pre-Python 3.13 Generator type syntax in conftest.py

Python 3.13 defaults to None for the other fields of the Generator type,
but the version of Generator in older versions does not support such syntax.
Explicitly add the missing fields to allow the tests to run on older Python
versions.

(An alternative solution would be to use typing-extensions for older
versions, though that would require adding another dependency.)

Signed-off-by: Artur Weber <aweber.kernel@gmail.com>
---
 tests/conftest.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/conftest.py b/tests/conftest.py
index c5ced78..a9c9157 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -20,7 +20,7 @@ This is a test email body.
 """
 
 @pytest.fixture
-def temp_data_dir() -> Generator[str]:
+def temp_data_dir() -> Generator[str, None, None]:
     """Create a temporary data directory structure for patatt."""
     with tempfile.TemporaryDirectory() as tmpdirname:
         # Create directory structure similar to patatt's data dir
-- 
2.52.0

